tensorflow/python/ops/variable_scope.py

```diff
@@ -690,6 +695,34 @@ def _get_single_variable(self,
     return v
 
 
+  # Initialize variable when no initializer provided
+  def _get_default_initializer(self, shape=None, dtype=dtypes.float32):
+    """Use this function to provide a default and corresponding initializer of dtype
+
+    Args:
+      dtype: Type of the new or existing variable.(defaults to `DT_FLOAT`).
+
+    Returns:
+      initializer and initializing_from_value. See documentation of get_variable above.
+
+    Raises:
+      TypeError: When giving unsupported dtype.
+    """
+    if dtype.is_floating:
+      initializer = init_ops.uniform_unit_scaling_initializer()
+      initializing_from_value = False
+    elif dtype.is_string:
+      initializer = init_ops.strings_initializer(shape=shape, dtype=dtype.base_dtype)
+      initializing_from_value=True
+    elif dtype.is_integer or dtype.is_unsigned or dtype.is_bool or dtype.is_complex:
```

Comment:
Yep, zeros initialization may not be the most appropriate if you just focus on NN. But I always    think tensorflow as a general Machine Learning platform, not only for NN, but also for other cases and solutions. As we can see the updated tutorials for [Non-ML Applications](https://www.tensorflow.org/versions/r0.11/tutorials/mandelbrot/index.html#mandelbrot-set), so I prefer to keep the initializer for complex.

Regarding zeros or sqrt(3), I think both are ok. Do you think so?